src/main/java/com/blackducksoftware/integration/hub/service/model/HubQuery.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * hub-common
+ *
+ * Copyright (C) 2018 Black Duck Software, Inc.
+ * http://www.blackducksoftware.com/
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.blackducksoftware.integration.hub.service.model;
+
+public class HubQuery {
+    private final String q;
+
+    public HubQuery(final String parameter) {
```

Comment:
Is there a time when 'parameter' won't be a key-value pair? I think we should have an alternate constructor that takes two strings (key and value), pairs them, and sets 'parameter' with the pairing. Let the string building happen here.